kernel/src/hil/flash.rs

```diff
@@ -125,3 +127,36 @@ pub trait Client<F: Flash> {
     /// Flash erase complete.
     fn erase_complete(&self, error: Error);
 }
+
+pub trait FlashInfo {
+    fn flash_size(&self) -> u32;
```

Comment:
Also, is this always num_pages * page_size? If so maybe there should be a default implementation.